mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -108,11 +111,19 @@ abstract class MLWriter extends BaseReadWrite with Logging {
   protected def saveImpl(path: String): Unit
 
   /**
-   * `option()` handles extra options. If subclasses need to support extra options, override this
-   * method.
+   * Map store extra options for this writer.
+   */
+  protected val optionMap: mutable.Map[String, String] = new mutable.HashMap[String, String]()
+
+  /**
+   * `option()` handles extra options.
```

Comment:
"Adds an option to the underlying MLWriter.  See the documentation for the specific model's writer for possible options.  The option name (key) is case-insensitive."